### ISSUE TITLE
Problem creating own image based on gvenzl/oracle-xe:11.2.0.2-faststart

### ISSUE THREAD

**Issue by hduelme** [gvenzl/oci-oracle-xe#222]
I want to create my own image based on `gvenzl/oracle-xe:11.2.0.2-faststart`.
I could start the container make my changes. 
Then I use `docker commit -p ` to create my own image.

But every time I try to start my own image I get

> CONTAINER: starting up...
CONTAINER: first database startup, initializing...
mkdir: cannot create directory '/u01/app/oracle/oradata': Permission denied

I assume the oracle thinks it has to create the database, but I already created one inside the container.

Any help would be greatly appreciated.

Thank you in advance for your time and support.

**Comment by hduelme** [comment]
This worked for me in the past

**Comment by gvenzl** [comment]
Hi @hduelme,

Thanks a lot for using these images!

I just ran through the steps to create a new image via `podman commit` and extending the image, both times the new image worked fine:

```
[gvenzl@localhost ~]$ podman ps -a
CONTAINER ID  IMAGE                                          COMMAND     CREATED         STATUS         PORTS       NAMES
ee83aa59d1b8  docker.io/gvenzl/oracle-xe:11.2.0.2-faststart              15 seconds ago  Up 16 seconds              test
[gvenzl@localhost ~]$ podman exec -ti test bash
bash-4.4$ sqlplus / as sysdba

SQL*Plus: Release 11.2.0.2.0 Production on Mon Aug 21 03:51:25 2023

Copyright (c) 1982, 2011, Oracle.  All rights reserved.


Connected to:
Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production

SQL> create user gerald identified by gerald;

User created.

SQL> shutdown immediate;
Database closed.
Database dismounted.
ORACLE instance shut down.
SQL> exit
Disconnected from Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production
bash-4.4$ exit
exit
[gvenzl@localhost ~]$ podman commit test newimage
WARN[0017] archive: skipping "/home/gvenzl/.local/share/containers/storage/overlay/261d0df0e5fee747ddce8b88542da158f4ed6d47909af029b637a8faa61b730f/diff/var/tmp/.oracle/s#20.1" since it is a socket
WARN[0017] archive: skipping "/home/gvenzl/.local/share/containers/storage/overlay/261d0df0e5fee747ddce8b88542da158f4ed6d47909af029b637a8faa61b730f/diff/var/tmp/.oracle/s#20.2" since it is a socket
WARN[0017] archive: skipping "/home/gvenzl/.local/share/containers/storage/overlay/261d0df0e5fee747ddce8b88542da158f4ed6d47909af029b637a8faa61b730f/diff/var/tmp/.oracle/sEXTPROC_FOR_XE" since it is a socket
Getting image source signatures
Copying blob 5f23a3c591bb skipped: already exists
Copying blob 2a1b5c9a8c9e skipped: already exists
Copying blob a7226642f719 skipped: already exists
Copying blob 343d7f283709 skipped: already exists
Copying blob 18d0c67aaf7f skipped: already exists
Copying blob fe7f6d18ab40 skipped: already exists
Copying blob d76894aa2954 skipped: already exists
Copying blob fbd17684f873 skipped: already exists
Copying blob c3f345e0e3a0 skipped: already exists
Copying blob af9965c64884 done
Copying config d61d459209 done
Writing manifest to image destination
Storing signatures
d61d4592090ac04ece03c1c3bd1e82acbfcf541a2e780617f5790785acb54589
[gvenzl@localhost ~]$ podman images
REPOSITORY                  TAG                 IMAGE ID      CREATED             SIZE
localhost/newimage          latest              d61d4592090a  About a minute ago  2.82 GB
localhost/test              latest              e9f4734f46db  5 minutes ago       2.82 GB
docker.io/gvenzl/oracle-xe  11.2.0.2-faststart  0923b720d7b7  5 weeks ago         1.73 GB
[gvenzl@localhost ~]$ podman ps -a
CONTAINER ID  IMAGE                                          COMMAND     CREATED        STATUS        PORTS       NAMES
ee83aa59d1b8  docker.io/gvenzl/oracle-xe:11.2.0.2-faststart              3 minutes ago  Up 3 minutes              test
[gvenzl@localhost ~]$ podman stop test
test
[gvenzl@localhost ~]$ podman run --name newcon -e ORACLE_RANDOM_PASSWORD=y newimage
CONTAINER: starting up...
CONTAINER: database already initialized.
CONTAINER: starting up Oracle Database...

LSNRCTL for Linux: Version 11.2.0.2.0 - Production on 21-AUG-2023 03:54:25

Copyright (c) 1991, 2011, Oracle.  All rights reserved.

Starting /u01/app/oracle/product/11.2.0/xe/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 11.2.0.2.0 - Production
System parameter file is /u01/app/oracle/product/11.2.0/xe/network/admin/listener.ora
Log messages written to /u01/app/oracle/diag/tnslsnr/3ce4973f5aff/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 11.2.0.2.0 - Production
Start Date                21-AUG-2023 03:54:25
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Default Service           XE
Listener Parameter File   /u01/app/oracle/product/11.2.0/xe/network/admin/listener.ora
Listener Log File         /u01/app/oracle/diag/tnslsnr/3ce4973f5aff/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
Services Summary...
Service "PLSExtProc" has 1 instance(s).
  Instance "PLSExtProc", status UNKNOWN, has 1 handler(s) for this service...
The command completed successfully
ORACLE instance started.

Total System Global Area  801701888 bytes
Fixed Size		    2230776 bytes
Variable Size		  222299656 bytes
Database Buffers	  574619648 bytes
Redo Buffers		    2551808 bytes
Database mounted.
Database opened.


#########################
DATABASE IS READY TO USE!
#########################

##################################################################
CONTAINER: The following output is now from the alert_XE.log file:
##################################################################
Database Characterset is AL32UTF8
Opening with Resource Manager plan: INTERNAL_PLAN_XE
Starting background process VKRM
Mon Aug 21 03:54:32 2023
VKRM started with pid=19, OS id=66
replication_dependency_tracking turned off (no async multimaster replication found)
Starting background process QMNC
Mon Aug 21 03:54:32 2023
QMNC started with pid=20, OS id=68
Completed: ALTER DATABASE OPEN
```

Could you please share the steps of how you created the image?

**Comment by hduelme** [comment]
Hi @gvenzl ,
I start container with `docker run -d -p 1521:1521 -e ORACLE_PASSWORD=test -e APP_USER=test -e APP_USER_PASSWORD=test --name oracle_own_test_db10 gvenzl/oracle-xe:11.2.0.2-faststart`.

Then I run my sql scripts.
After this I stop my container by running `docker stop oracle_own_test_db10`.

Here the logs of the stopped container.
```
CONTAINER: starting up...
CONTAINER: first database startup, initializing...
CONTAINER: starting up Oracle Database...

LSNRCTL for Linux: Version 11.2.0.2.0 - Production on 21-AUG-2023 07:04:03

Copyright (c) 1991, 2011, Oracle.  All rights reserved.

Starting /u01/app/oracle/product/11.2.0/xe/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 11.2.0.2.0 - Production
System parameter file is /u01/app/oracle/product/11.2.0/xe/network/admin/listener.ora
Log messages written to /u01/app/oracle/diag/tnslsnr/d28f195a9cef/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 11.2.0.2.0 - Production
Start Date                21-AUG-2023 07:04:03
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Default Service           XE
Listener Parameter File   /u01/app/oracle/product/11.2.0/xe/network/admin/listener.ora
Listener Log File         /u01/app/oracle/diag/tnslsnr/d28f195a9cef/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
Services Summary...
Service "PLSExtProc" has 1 instance(s).
  Instance "PLSExtProc", status UNKNOWN, has 1 handler(s) for this service...
The command completed successfully
ORACLE instance started.

Total System Global Area  801701888 bytes
Fixed Size		    2230776 bytes
Variable Size		  222299656 bytes
Database Buffers	  574619648 bytes
Redo Buffers		    2551808 bytes
Database mounted.
Database opened.

CONTAINER: Resetting SYS and SYSTEM passwords.

User altered.


User altered.


User created.


Grant succeeded.


#########################
DATABASE IS READY TO USE!
#########################

##################################################################
CONTAINER: The following output is now from the alert_XE.log file:
##################################################################
Database Characterset is AL32UTF8
Opening with Resource Manager plan: INTERNAL_PLAN_XE 
Starting background process VKRM
Mon Aug 21 07:04:22 2023
VKRM started with pid=19, OS id=72 
replication_dependency_tracking turned off (no async multimaster replication found)
Starting background process QMNC
Mon Aug 21 07:04:22 2023
QMNC started with pid=20, OS id=74 
Completed: ALTER DATABASE OPEN
Starting background process CJQ0
Mon Aug 21 07:04:23 2023
CJQ0 started with pid=23, OS id=109 
Mon Aug 21 07:04:26 2023
Errors in file /u01/app/oracle/diag/rdbms/xe/XE/trace/XE_j002_115.trc:
ORA-12012: error on auto execute of job "SYS"."BSLN_MAINTAIN_STATS_JOB"
ORA-06550: line 1, column 807:
PLS-00201: identifier 'DBSNMP.BSLN_INTERNAL' must be declared
ORA-06550: line 1, column 807:
PL/SQL: Statement ignored
CONTAINER: shutdown request received.
CONTAINER: shutting down database!

LSNRCTL for Linux: Version 11.2.0.2.0 - Production on 21-AUG-2023 07:05:48

Copyright (c) 1991, 2011, Oracle.  All rights reserved.

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
The command completed successfully
Mon Aug 21 07:05:48 2023
Shutting down instance (immediate)
Shutting down instance: further logons disabled
Mon Aug 21 07:05:49 2023
Stopping background process CJQ0
Stopping background process QMNC
Stopping background process MMNL
Stopping background process MMON
License high water mark = 4
All dispatchers and shared servers shutdown
ALTER DATABASE CLOSE NORMAL
Mon Aug 21 07:05:52 2023
SMON: disabling tx recovery
SMON: disabling cache recovery
Mon Aug 21 07:05:52 2023
Shutting down archive processes
Archiving is disabled
Archive process shutdown avoided: 0 active
Thread 1 closed at log sequence 14
Successful close of redo thread 1
Completed: ALTER DATABASE CLOSE NORMAL
Database closed.
ALTER DATABASE DISMOUNT
Completed: ALTER DATABASE DISMOUNT
Database dismounted.
ARCH: Archival disabled due to shutdown: 1089
Shutting down archive processes
Archiving is disabled
ARCH: Archival disabled due to shutdown: 1089
Shutting down archive processes
Archiving is disabled
Mon Aug 21 07:05:53 2023
Stopping background process VKTM
Mon Aug 21 07:05:55 2023
Instance shutdown complete
ORACLE instance shut down.
CONTAINER: stopping container.
```

Now I build my image by running `docker commit -p oracle_own_test_db10 oracle_own_test_db`.
When I try to run my own image I get:
```
docker run oracle_own_test_db
CONTAINER: starting up...
CONTAINER: first database startup, initializing...
mkdir: cannot create directory '/u01/app/oracle/oradata': Permission denied
```

**Comment by hduelme** [comment]
I makes no difference if I use `docker run -d -p 1521:1521 -e ORACLE_PASSWORD=test -e APP_USER=test -e APP_USER_PASSWORD=test oracle_own_test_db` to start the container

**Comment by gvenzl** [comment]
Hey @hduelme,

Hm, this is strange, I just followed the same steps and it worked:

```
[gvenzl@localhost ~]$ podman run -d -p 1521:1521 -e ORACLE_PASSWORD=test -e APP_USER=test -e APP_USER_PASSWORD=test --name oracle_own_test_db10 gvenzl/oracle-xe:11.2.0.2-faststart
Resolving "gvenzl/oracle-xe" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull docker.io/gvenzl/oracle-xe:11.2.0.2-faststart...
Getting image source signatures
Copying blob f285b3c37ad8 done
Copying blob dfbf381a1261 done
Copying blob a359813b8cb8 done
Copying blob 06e3c5b9c861 done
Copying blob 4123ea1421fc done
Copying blob b6b4836ff0a3 done
Copying blob 92a3afa67841 done
Copying blob 576dff46653b done
Copying blob 50f6d35eaafe done
Copying config 0923b720d7 done
Writing manifest to image destination
Storing signatures
613908408e2f33f0ae98a4f21edb937662ed9d2baa97c184f619ba6ad5119e7c
[gvenzl@localhost ~]$ podman exec -ti oracle_own_test_db10 bash
bash-4.4$ id
uid=1000(oracle) gid=1000(dba) groups=1000(dba)
bash-4.4$ sqlplus / as sysdba

SQL*Plus: Release 11.2.0.2.0 Production on Sun Sep 10 03:20:36 2023

Copyright (c) 1982, 2011, Oracle.  All rights reserved.


Connected to:
Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production

SQL> conn test/test
Connected.
SQL> create table foo (id int);

Table created.

SQL> insert into foo values (1);

1 row created.

SQL> insert into foo values (2);

1 row created.

SQL> commit;

Commit complete.

SQL> exit
Disconnected from Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production
bash-4.4$ exit
exit
[gvenzl@localhost ~]$ podman stop oracle_own_test_db10
oracle_own_test_db10
[gvenzl@localhost ~]$ podman commit -p oracle_own_test_db10 oracle_own_test_db
Getting image source signatures
Copying blob 5f23a3c591bb skipped: already exists
Copying blob 2a1b5c9a8c9e skipped: already exists
Copying blob a7226642f719 skipped: already exists
Copying blob 343d7f283709 skipped: already exists
Copying blob 18d0c67aaf7f skipped: already exists
Copying blob fe7f6d18ab40 skipped: already exists
Copying blob d76894aa2954 skipped: already exists
Copying blob fbd17684f873 skipped: already exists
Copying blob c3f345e0e3a0 skipped: already exists
Copying blob 324c791ef775 done
Copying config a1ee2b7b93 done
Writing manifest to image destination
Storing signatures
a1ee2b7b938a2ecebeebd980fd1204bd4e5beae1726d5ba41322fb8ec81ea116
[gvenzl@localhost ~]$ podman images
REPOSITORY                    TAG                 IMAGE ID      CREATED        SIZE
localhost/oracle_own_test_db  latest              a1ee2b7b938a  4 minutes ago  2.82 GB
docker.io/gvenzl/oracle-xe    11.2.0.2-faststart  0923b720d7b7  2 months ago   1.73 GB
[gvenzl@localhost ~]$ podman run localhost/oracle_own_test_db
CONTAINER: starting up...
CONTAINER: database already initialized.
CONTAINER: starting up Oracle Database...

LSNRCTL for Linux: Version 11.2.0.2.0 - Production on 10-SEP-2023 03:26:19

Copyright (c) 1991, 2011, Oracle.  All rights reserved.

Starting /u01/app/oracle/product/11.2.0/xe/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 11.2.0.2.0 - Production
System parameter file is /u01/app/oracle/product/11.2.0/xe/network/admin/listener.ora
Log messages written to /u01/app/oracle/diag/tnslsnr/3645492e83a7/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 11.2.0.2.0 - Production
Start Date                10-SEP-2023 03:26:19
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Default Service           XE
Listener Parameter File   /u01/app/oracle/product/11.2.0/xe/network/admin/listener.ora
Listener Log File         /u01/app/oracle/diag/tnslsnr/3645492e83a7/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
Services Summary...
Service "PLSExtProc" has 1 instance(s).
  Instance "PLSExtProc", status UNKNOWN, has 1 handler(s) for this service...
The command completed successfully
ORACLE instance started.

Total System Global Area  801701888 bytes
Fixed Size		    2230776 bytes
Variable Size		  222299656 bytes
Database Buffers	  574619648 bytes
Redo Buffers		    2551808 bytes
Database mounted.
Database opened.

CONTAINER: WARNING: $ORACLE_PASSWORD has been specified but the database is already initialized. The password will be ignored.
CONTAINER: WARNING: If you want to reset the password, please run the resetPassword command, e.g. 'docker|podman exec <container name|id> resetPassword <your password>'.

#########################
DATABASE IS READY TO USE!
#########################

##################################################################
CONTAINER: The following output is now from the alert_XE.log file:
##################################################################
Database Characterset is AL32UTF8
Opening with Resource Manager plan: INTERNAL_PLAN_XE
Starting background process VKRM
Sun Sep 10 03:26:26 2023
VKRM started with pid=19, OS id=66
replication_dependency_tracking turned off (no async multimaster replication found)
Starting background process QMNC
Sun Sep 10 03:26:26 2023
QMNC started with pid=20, OS id=68
Completed: ALTER DATABASE OPEN
^Z^CCONTAINER: shutdown request received.
CONTAINER: shutting down database!

LSNRCTL for Linux: Version 11.2.0.2.0 - Production on 10-SEP-2023 03:26:36

Copyright (c) 1991, 2011, Oracle.  All rights reserved.

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
The command completed successfully
Sun Sep 10 03:26:37 2023
Shutting down instance (immediate)
Shutting down instance: further logons disabled
Stopping background process QMNC
Stopping background process MMNL
Stopping background process MMON
Database closed.
Database dismounted.
License high water mark = 1
All dispatchers and shared servers shutdown
ALTER DATABASE CLOSE NORMAL
Sun Sep 10 03:26:40 2023
SMON: disabling tx recovery
SMON: disabling cache recovery
Sun Sep 10 03:26:40 2023
Shutting down archive processes
Archiving is disabled
Archive process shutdown avoided: 0 active
Thread 1 closed at log sequence 14
Successful close of redo thread 1
Completed: ALTER DATABASE CLOSE NORMAL
ALTER DATABASE DISMOUNT
Completed: ALTER DATABASE DISMOUNT
ARCH: Archival disabled due to shutdown: 1089
Shutting down archive processes
Archiving is disabled
ARCH: Archival disabled due to shutdown: 1089Sun Sep 10 03:26:40 2023

Stopping background process VKTM
Shutting down archive processes
Archiving is disabled
ORACLE instance shut down.
CONTAINER: stopping container.
[gvenzl@localhost ~]$
```

Are the scripts doing anything on the file system at all?
